include/boost/geometry/algorithms/buffer.hpp

```diff
@@ -242,22 +242,23 @@ inline void buffer(GeometryIn const& geometry_in,
     geometry::envelope(geometry_in, box);
     geometry::buffer(box, box, distance_strategy.max_distance(join_strategy, end_strategy));
 
-    typename strategy::intersection::services::default_strategy
+    typename strategies::relate::services::default_strategy
         <
-            typename cs_tag<GeometryIn>::type
-        >::type intersection_strategy;
+            GeometryIn, GeometryIn
+        >::type strategy;
```

Comment:
So buffer uses the default relate umbrella strategy instead of intersection strategy.

Right now buffer strategies are at the same time parameters/options for buffer e.g. holding distance value and CS-specific strategies defining how parts of buffer are calculated. I propose to separate that in the future.

So this would be a separate buffer strategy passed by the user, also defining how specific buffer-related things are calculated (like generation of new points in various coordinate systems). And then I think `buffer()` algorithm should take parameters/options specifying distance/line-end/etc. plus one optional CS-specific umbrella strategy.